SwiftLSPClient/Types/Basic.swift

```diff
@@ -67,8 +67,20 @@ public enum LanguageIdentifier: String, Codable, CaseIterable {
         "cpp": .cpp,
         "m": .objc,
         "mm": .objcpp,
-        "h": .objcpp
     ]
+
+    public struct UnsupportedFileType: Error {
+        let msg: String
+    }
```

Comment:
I was thinking about this more. There are actually a bunch of other kinds of parameter validation that could be done. Perhaps instead we can do something like this:

    public enum LanguageServerParameterError: Error {
        case unsupportedFileExtension(String)
    }

This allows for much better communication of the semantics in the error, while also making it possible to expose all kinds of other parameter errors in the future without custom types for each one. This is a pretty common idiom for swift throwing errors, and it's already in use in the library with `LanguageServerError`. You into that?

Reply:
Done 👍